sdk/src/metrics/state/temporal_metric_storage.cc

```diff
@@ -51,6 +51,35 @@ bool TemporalMetricStorage::buildMetrics(CollectorHandle *collector,
   AggregationTemporality aggregation_temporarily =
       collector->GetAggregationTemporality(instrument_descriptor_.type_);
 
+  // Fast path for single collector with delta temporality and counter, updown-counter, histogram
+  // This path doesn't need to aggregated-with/contribute-to the unreported_metric_, as there is
+  // no other reader configured to collect those data.
+  if (collectors.size() == 1 && aggregation_temporarily == AggregationTemporality::kDelta)
```

Comment:
Short circuit return if `collectors.size() == 0`?

Reply:
Yes, can be done, but this is separate of the current issue handled? Will add it as separate PR.